ql/src/java/org/apache/hadoop/hive/ql/security/authorization/plugin/metastore/HiveMetaStoreAuthorizer.java

```diff
@@ -213,15 +214,46 @@ public final List<String> filterCatalogs(List<String> catalogs) throws MetaExcep
 
   @Override
   @Deprecated
-  public List<TableMeta> filterTableMetas(String catName, String dbName,List<TableMeta> tableMetas)
+  public List<TableMeta> filterTableMetas(String catName, String dbName, List<TableMeta> tableMetas)
       throws MetaException {
-    return filterTableMetas(tableMetas);
+    if (LOG.isDebugEnabled()) {
+      LOG.debug("==> HiveMetaStoreAuthorizer.filterTableMetas()");
+    }
+    List<String> tableNames = new ArrayList<>();
+    List<TableMeta> filteredTableMetas = new ArrayList<>();
+    if (tableMetas != null) {
+      for (TableMeta tableMeta : tableMetas) {
+        tableNames.add(tableMeta.getTableName());
+      }
+      TableFilterContext     tableFilterContext     = new TableFilterContext(dbName, tableNames);
+      HiveMetaStoreAuthzInfo hiveMetaStoreAuthzInfo = tableFilterContext.getAuthzContext();
+      final List<String>  filteredTableNames = filterTableNames(hiveMetaStoreAuthzInfo, dbName, tableNames);
+      if (CollectionUtils.isEmpty(filteredTableNames)) {
+        filteredTableMetas = Collections.emptyList();
+        if (LOG.isInfoEnabled()) {
+          LOG.info("<== HiveMetaStoreAuthorizer.filterTableMetas() : returning empty set");
+        }
+      } else {
+        if (LOG.isDebugEnabled()) {
+          LOG.debug("<== HiveMetaStoreAuthorizer.filterTableMetas() : " + filteredTableNames);
+        }
+        filteredTableMetas = tableMetas.stream().filter(tblMeta -> filteredTableNames.stream()
+            .anyMatch(tblName -> tblName.equals(tblMeta.getTableName()))).collect(Collectors.toList());
+      }
+    }
+    return filteredTableMetas;
   }
 
   @Override
   public final List<TableMeta> filterTableMetas(List<TableMeta> tableMetas)
       throws MetaException {
-    return tableMetas;
+    String catName = null;
+    String dbName = null;
+    if (tableMetas != null) {
+      catName = tableMetas.get(0).getCatName();
+      dbName = tableMetas.get(0).getDbName();
```

Comment:
I am not sure about this, but will the "tableMetas" have only single DbName? Can't it have, multiple DbNames?

Reply:
sounds like we might get multiple dbs in this call, as the param `dbnames` in `get_table_meta` is a pattern.

Reply:
So, I think in the other method, sending same dbName with the list of tables in "TableFilterContext" would not work as expected.